### PR TITLE
librbd/migration: add thick-provisioned snapshot support to the raw format

### PR DESCRIPTION
**Pull request by dillaman**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by trociny — APPROVED** [review]
LGTM

File: qa/workunits/rbd/cli_migration.sh [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    done
+}
+
+compare_images() {
+    local src_image=$1
+    local dst_image=$2
+    local ret=0
+
+    export_raw_image ${dst_image}
+    if ! cmp "${TEMPDIR}/${src_image}" "${TEMPDIR}/${dst_image}"
```
[the remark this answers is not in the document]
Comment (dillaman) [reply qa/workunits/rbd/cli_migration.sh:72]: Sure -- I just copied-and-pasted it w/ some simplifications

**Comment by trociny** [comment]
@dillaman The rbd_cli_migration tests may be reported as failed due to "slow request" warnings [1]. It seems `rbd bench` the test runs creates considerable load for such a tiny cluster that is used for this test.

Might we want to improve this? E.g. decrease the bench load (it runs for rather long time on that cluster, while probably a much smaller number of updates would be enough for the test) or filter out slow op warnings? Or both.

[1] https://pulpito.ceph.com/trociny-2020-12-03_16:25:16-rbd-wip-mgolub-testing-distro-basic-smithi/

**Comment by dillaman** [comment]
The `cli_generic` test hits it as well on the small cache-pool tests -- maybe the tiering tests should just get scaled out (as a separate PR)

**Comment by dillaman** [comment]
jenkins test make check

**Comment by dillaman** [comment]
jenkins test api

**Comment by dillaman** [comment]
jenkins test make check

**Pull request merged**